src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs

```diff
@@ -0,0 +1,119 @@
+
+        public TeamController(TeamService teamservice, IMapper mapper)
+        {
+            _teamService = teamservice;
+            _mapper = mapper;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TeamRead>> GetTeamById(int id)
+        {
+            var team = await _teamService.GetTeamById(id);
+            if(team==null)
+            {
+                return NotFound();
+            }
+
+            var teamRead = _mapper.Map<TeamRead>(team);
+            return Ok(teamRead);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TeamRead>>> GetAllTeams()
+        {
+            var teams = await _teamService.GetAllTeams();
+            var teamRead = _mapper.Map<IEnumerable<TeamRead>>(teams);
+            return Ok(teamRead);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<TeamRead>> AddTeam(TeamCreate teamCreate)
+        {
+            var team = _mapper.Map<Team>(teamCreate);
+            await _teamService.AddTeam(team);
+            var teamRead = _mapper.Map<TeamRead>(team);
+            return CreatedAtAction(nameof(GetTeamById), new { id = teamRead.Id }, teamRead);
+        }
+
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateTeam(int id, TeamUpdate teamUpdate)
+        {
+            var team = _mapper.Map<Team>(teamUpdate);
+            team.Id = id;
+
+            try
+            {
+                await _teamService.UpdateTeam(team);
+            }
+            catch(DbUpdateConcurrencyException)
+            {
+                if(await _teamService.GetTeamById(id) == null)
+                {
+                    return NotFound();
+                }
+
+                throw;
+            }
+
+            return NoContent();
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteTeam(int id)
+        {
+            var team = await _teamService.GetTeamById(id);
+            if(team == null)
+            {
+                return NotFound();
+            }
+
+            await _teamService.DeleteTeam(team);
+
+            return NoContent();
+        }
+
+        [HttpPost("{teamId}/players")]
+        public async Task<ActionResult<ApplicationUser>> AddPlayer(int teamId, ApplicationUser player)
+        {
+            await _teamService.AddPlayer(teamId, player);
+            return CreatedAtAction(nameof(GetPlayerById), new { teamId, id = player.Id }, player);
+        }
+
+        [HttpGet("{teamId}/players/{id}")]
+        public async Task<ActionResult<ApplicationUser>> GetPlayerById(int teamId, int id)
+        {
+            var team = await _teamService.GetTeamById(teamId);
+            if(team == null)
+            {
+                return NotFound();
+            }
+
+            var player = team.Players.FirstOrDefault(p => Convert.ToInt32(p.Id) == id);
+            if(player==null)
+            {
+                return NotFound(); ;
+            }
```

Comment:
Most of this logic could be in the service. Also I think it would be better to throw an Exception (in the service) that inherits from BaseDomainException and then you could do something like this in the controller:
````
try {
 // Do stuff
} catch (NotFoundException) {
 return NotFound();
}
````

Or if possible you could catch that in the GlobalExceptionMiddleware and that way we would not need to return NotFound() anywhere else in the code. Although I am not sure if that middleware catches non async methods.